lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -240,7 +240,7 @@ def get_cpu_facts(self, collected_facts=None):
         # The fields for ARM CPUs do not always include 'vendor_id' or 'model name',
         # and sometimes includes both 'processor' and 'Processor'.
         # Always use 'processor' count for ARM systems
-        if collected_facts.get('ansible_architecture').startswith(('armv', 'aarch')):
+        if collected_facts.get('ansible_architecture', '').startswith(('armv', 'aarch')):
```

Comment:
you would still need to change 2 lines down

Reply:
also the other files in facts that make the same assumptions

Reply:
The comparison two lines down is safe because it will default to `None`.

https://github.com/ansible/ansible/blob/d85757c89264522720a27602d37ca5590d695397/lib/ansible/module_utils/facts/hardware/linux.py#L247

I looked through the facts code and most instances fail safe, meaning they don't do things that would result in an exception like I did on line 243 (`None` has not `startswith()` method. Doh!).